lib/dialects/sqlite/query.js

```diff
@@ -1,385 +1,371 @@
+        let key;
+        if (!prefix) {
+          key = _include.as;
+        } else {
+          key = prefix + '.' + _include.as;
+        }
+        ret[key] = _include.model;
 
-Query.prototype.run = function(sql, parameters) {
-  var self = this
-    , promise;
+        if (_include.include) {
+          _.merge(ret, this.$collectModels(_include.include, key));
+        }
+      }
+    }
 
-  this.sql = sql;
-  var method = self.getDatabaseMethod();
-  if (method === 'exec') {
-    // exec does not support bind parameter
-    sql = AbstractQuery.formatBindParameters(sql, self.options.bind, self.options.dialect, { skipUnescape: true })[0];
-    this.sql = sql;
+    return ret;
   }
 
-  //do we need benchmark for this query execution
-  var benchmark = this.sequelize.options.benchmark || this.options.benchmark;
+  run(sql, parameters) {
+    this.sql = sql;
+    const method = this.getDatabaseMethod();
+    if (method === 'exec') {
+      // exec does not support bind parameter
+      sql = AbstractQuery.formatBindParameters(sql, this.options.bind, this.options.dialect, { skipUnescape: true })[0];
+      this.sql = sql;
+    }
 
-  if (benchmark) {
-    var queryBegin = Date.now();
-  } else {
-    this.sequelize.log('Executing (' + (this.database.uuid || 'default') + '): ' + this.sql, this.options);
-  }
+    //do we need benchmark for this query execution
+    const benchmark = this.sequelize.options.benchmark || this.options.benchmark;
 
-  promise = new Promise(function(resolve) {
-    var columnTypes = {};
-    self.database.serialize(function() {
-      var executeSql = function() {
-        if (self.sql.indexOf('-- ') === 0) {
-          return resolve();
-        } else {
-          resolve(new Promise(function(resolve, reject) {
-            var afterExecute = function(err, results) {
+    let queryBegin;
+    if (benchmark) {
+      queryBegin = Date.now();
+    } else {
+      this.sequelize.log('Executing (' + (this.database.uuid || 'default') + '): ' + this.sql, this.options);
+    }
 
-              if (benchmark) {
-                self.sequelize.log('Executed (' + (self.database.uuid || 'default') + '): ' + self.sql, (Date.now() - queryBegin), self.options);
-              }
+    return new Promise(resolve => {
+      const columnTypes = {};
+      this.database.serialize(() => {
+        const executeSql = () => {
+          if (this.sql.indexOf('-- ') === 0) {
+            return resolve();
+          } else {
+            resolve(new Promise((resolve, reject) => {
+              const afterExecute = (err, results) => {
+
+                if (benchmark) {
+                  this.sequelize.log('Executed (' + (this.database.uuid || 'default') + '): ' + this.sql, (Date.now() - queryBegin), this.options);
+                }
 
-              if (err) {
-                err.sql = self.sql;
-                reject(self.formatError(err));
-              } else {
-                var metaData = this
```

Comment:
`this` here previously referred to the statement object, which contains the last insert id. So I don't think we can use an array function here (unless you can find another way to get the statement object).

https://github.com/mapbox/node-sqlite3/wiki/API

> callback (optional): If given, it will be called when an error occurs during any step of the statement preparation or execution, and after the query was run. If an error occurred, the first (and only) parameter will be an error object containing the error message. If execution was successful, the first parameter is null. The context of the function (the this object inside the function) is the statement object. Note that it is not possible to run the statement again because it is automatically finalized after running for the first time. Any subsequent attempts to run the statement again will fail.